ibis/formats/pyarrow.py

```diff
@@ -12,6 +13,34 @@
 if TYPE_CHECKING:
     from collections.abc import Sequence
 
+
+class JSONScalar(pa.ExtensionScalar):
+    def as_py(self):
+        return json.loads(self.value.as_py())
```

Comment:
The `json.loads` call is the meat of this PR, and it's what gets displayed in the rich table we produce.